modules/network/src/test/java/org/apache/ignite/network/MessageSerializationRegistryImplTest.java

```diff
@@ -45,11 +47,34 @@ public void testRegisterFactory() {
      */
     @Test
     public void testRegisterFactoryWithSameType() {
-        var registry = new MessageSerializationRegistry();
+        var registry = new MessageSerializationRegistryImpl();
 
-        registry.registerFactory(Msg.TYPE, new MsgSerializationFactory());
+        registry.registerFactory(Msg.GROUP_TYPE, Msg.TYPE, new MsgSerializationFactory());
 
-        assertThrows(NetworkConfigurationException.class, () -> registry.registerFactory(Msg.TYPE, new MsgSerializationFactory()));
+        assertThrows(
+            NetworkConfigurationException.class,
+            () -> registry.registerFactory(Msg.GROUP_TYPE, Msg.TYPE, new MsgSerializationFactory())
+        );
+    }
+
+    /**
+     * Tests that it is possible to register serialization factories for the same message types but for different
+     * modules.
+     */
+    @Test
+    public void testRegisterFactoryWithSameTypeDifferentModule() {
```

Comment:
What a weird test. Do we have a similar test that checks that deserializers are different for the same type? Same for serializers (they're stateful, right?)

Reply:
Nice catch again, the assertion is indeed useless (what a shame)